src/coreclr/src/vm/eventtrace.cpp

```diff
@@ -3467,21 +3467,20 @@ BOOL ETW::TypeSystemLog::AddTypeToGlobalCacheIfNotExists(TypeHandle th, BOOL * p
 
     BOOL fSucceeded = FALSE;
 
+    // Check if ETW is enabled, and if not, bail here.
+    // We do this inside of the lock to ensure that we don't immediately
+    // re-allocate the global type hash after it has been cleaned up.
+    if (!ETW_TRACING_CATEGORY_ENABLED(
```

Comment:
I think it is safe to leave this inside the lock where it was. Did you observe a problem that was causing?